release/benchmarks/distributed/dashboard_test.py

```diff
@@ -0,0 +1,144 @@
+import asyncio
+import time
+from typing import Dict, Optional
+from pprint import pprint
+
+import requests
+import ray
+import logging
+
+from collections import defaultdict
+from ray._private.test_utils import format_web_url, fetch_prometheus_metrics
+from ray.util.scheduling_strategies import NodeAffinitySchedulingStrategy
+from pydantic import BaseModel
+from ray.dashboard.consts import DASHBOARD_METRIC_PORT
+
+logger = logging.getLogger(__name__)
+
+
+class Result(BaseModel):
+    success: bool
+    # endpoints -> P95 latency
+    p95_ms: Optional[Dict[str, float]]
+    # endpoints -> P95 latency
+    p50_ms: Optional[Dict[str, float]]
+    # Dashboard memory usage in MB.
+    memory_mb: Optional[float]
+
+
+# Currently every endpoint is GET endpoints.
+endpoints = [
+    "/logical/actors",
+    "/nodes?view=summary",
+    "/",
+    "/api/cluster_status",
+    "/events",
+    "/api/jobs/",
+    "/log_index",
+    "/api/prometheus_health",
+]
+
+
+@ray.remote(num_cpus=0)
+class DashboardTester:
+    def __init__(self, addr: ray._private.worker.RayContext, interval_s: int = 1):
+        self.dashboard_url = format_web_url(addr.dashboard_url)
```

Comment:
note: I found this doesn't work at Anyscale because they change the dashboard URL. I am working on a fix